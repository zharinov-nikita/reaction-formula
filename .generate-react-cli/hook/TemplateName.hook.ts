import { UseTemplateName, UseTemplateNameProps } from './template-name.types'

export function useTemplateName(props: UseTemplateNameProps): UseTemplateName {
  return {}
}
