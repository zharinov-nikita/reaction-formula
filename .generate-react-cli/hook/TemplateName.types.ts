export interface UseTemplateNameProps {}
export interface UseTemplateName {}
