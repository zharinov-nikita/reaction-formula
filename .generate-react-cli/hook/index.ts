export * from './template-name.hook'
export * from './template-name.types'