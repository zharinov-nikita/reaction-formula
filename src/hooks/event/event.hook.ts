import { useState } from 'react'
import { TypeEvent, TypeEventHandler } from '../../types'
import { UseEvent } from './event.types'

export function useEvent(): UseEvent {
  const [isFocus, setIsFocus] = useState<boolean>(false)
  const [isBlur, setIsBlur] = useState<boolean>(false)

  function onFocus() {
    setIsFocus(true)
    setIsBlur(false)
  }

  function onBlur() {
    setIsBlur(true)
    setIsFocus(false)
  }

  const eventHandler: TypeEventHandler = {
    onFocus,
    onBlur,
  }

  const event: TypeEvent = {
    isBlur,
    isFocus,
  }

  return { eventHandler, event }
}
