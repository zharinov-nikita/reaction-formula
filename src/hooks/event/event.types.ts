import { EventHandlerType, EventType } from '../../types'

export interface UseEvent {
  eventHandler: EventHandlerType
  event: EventType
}
