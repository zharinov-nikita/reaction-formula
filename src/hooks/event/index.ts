export * from './event.hook'
export * from './event.types'
