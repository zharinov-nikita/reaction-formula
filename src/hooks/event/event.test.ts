import { useEvent } from './event.hook'

describe('useEvent', () => {
  test('snapshot', () => {
    expect(useEvent).toMatchSnapshot('snapshot')
  })
})
