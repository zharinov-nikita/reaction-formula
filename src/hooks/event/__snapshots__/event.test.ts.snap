// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useEvent snapshot: snapshot 1`] = `[Function]`;
